src/connection.rs

```diff
@@ -0,0 +1,127 @@
+use std::collections::HashMap;
+use std::net::{AddrParseError, IpAddr, Ipv4Addr, Ipv6Addr};
+use std::str::{FromStr};
+use std::string::ToString;
+use std::error::Error;
+use std::time::{Duration, Instant};
+use std::default::Default;
+
+use amethyst_error::AmethystNetworkError;
+
+
+/// Type aliases
+/// Number of seconds we will wait until we consider a Connection to have timed out
+type ConnectionTimeout = u8;
+/// The port associated with this Connection
+type NetworkPort = u16;
+
+/// Default timeout of 10 seconds
+const TIMEOUT_DEFAULT: ConnectionTimeout = 10;
+
+
+/// Maintains a list of all Connections and allows adding/removing them
+#[derive(Default)]
+pub struct Manager {
+    /// The collection of currently connected clients
+    connections: HashMap<String, Connection>,
+    /// The number of seconds before we consider a client to have timed out
+    timeout: ConnectionTimeout
+}
+
+impl Manager {
+    pub fn new() -> Manager {
+        Manager {
+            connections: HashMap::new(),
+            timeout: TIMEOUT_DEFAULT
+        }
+    }
+
+    pub fn with_client_timeout(mut self, timeout: ConnectionTimeout) -> Manager {
+        self.timeout = timeout;
+        self
+    }
+
+    /// Adds a new connection to the manager.
+    pub fn add_connection(&mut self, conn: Connection) -> Result<(), AmethystNetworkError> {
+        if !self.connections.contains_key(&conn.to_string()) {
+            self.connections.insert(conn.to_string(), conn);
+            Ok(())
+        } else {
+            Err(AmethystNetworkError::AddConnectionToManagerFailed{reason: "Entry already exists".to_string()})
+        }
+    }
+}
+
+/// Represents a virtual circuit to a remote endpoint
+pub struct Connection {
+    /// IP Address of the remote endpoint
+    remote_ip: IpAddr,
```

Comment:
I would change this to use `SocketAddr` as you can store both the port and ip in the same type.